src/util/adb.js

```diff
@@ -210,10 +210,45 @@ export default class ADBUtils {
     await this.runShellCommand(deviceId, ['mkdir', '-p', artifactsDir]);
 
     this.artifactsDirMap.set(deviceId, artifactsDir);
-
     return artifactsDir;
   }
 
+  async checkOrCleanArtifacts(
+    deviceId: string, clean: boolean): Promise<boolean> {
+    const {adbClient} = this;
+
+    var directories = false;
+
+    await wrapADBCall(async () => {
+      log.debug('Finding Older Artifacts');
+
+      const dirs = await adbClient.readdir(deviceId, '/sdcard');
```

Comment:
lets not call this `dirs`, as it is misleading, this is not going to contains only directories. `files` may be a better name.